pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -710,12 +713,32 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *api.Serv
 		glog.Infof("EnsureLoadBalancer(%v(%v)): deleted forwarding rule", loadBalancerName, serviceName)
 	}
 	if tpExists && tpNeedsUpdate {
-		if err := gce.deleteTargetPool(loadBalancerName, gce.region); err != nil {
+		hc := []*compute.HttpHealthCheck{}
+		if path, healthCheckNodePort := getServiceHealthCheckPathPort(apiService); path != "" {
+			name := fmt.Sprintf("%v-%d", loadBalancerName, healthCheckNodePort)
+			c, err := gce.GetHttpHealthCheck(name)
+			if err != nil && !isHTTPErrorCode(err, http.StatusNotFound) {
+				glog.Infof("Failed to retrieve health check %v:%v", name, err)
+			} else {
+				hc = append(hc, c)
```

Comment:
this list is used to delete health checks, deleting a non-existent health check should no-op (https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce.go#L1785), though it is one wasted round trip to the cloud.